### PR TITLE
[INLONG-4796][Manager] Support management of Inlong cluster tag

### LINKED ISSUE #4796: [Feature][Manager] Support management of Inlong cluster tag
### Description

Support management of Inlong cluster tag.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by healchow**
### Prepare a Pull Request

- Fixes #4796

### Motivation

Support management of Inlong cluster tag.

### Modifications

1. Add APIs in InlongClusterController.
2. Add inlong_cluster_tag table.

### Verifying this change

- [x] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java [hunk 1]
```diff
@@ -83,19 +88,111 @@ public class InlongClusterServiceImpl implements InlongClusterService {
...
     private InlongClusterEntityMapper clusterMapper;
     @Autowired
     private InlongClusterNodeEntityMapper clusterNodeMapper;
     @Autowired
     private DataProxyConfigRepository proxyRepository;
 
+    @Override
+    public Integer saveTag(ClusterTagRequest request, String operator) {
+        LOGGER.debug("begin to save cluster tag {}", request);
+        Preconditions.checkNotNull(request, "cluster tag cannot be empty");
```
Comment (kipshi) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:102]: May be check clusterTag is not empty , or request empty?

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java [hunk 2]
```diff
@@ -47,14 +50,53 @@
  * Inlong cluster controller
  */
 @RestController
-@RequestMapping("/cluster")
```
Comment (kipshi) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:50]: why remove path identify in controller

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java [hunk 3]
```diff
@@ -83,19 +88,111 @@ public class InlongClusterServiceImpl implements InlongClusterService {
...
+        InlongClusterTagEntity exist = clusterTagMapper.selectByTag(clusterTag);
+        if (exist != null) {
+            String errMsg = String.format("inlong cluster tag already exist for cluster tag=%s", clusterTag);
+            LOGGER.error(errMsg);
+            throw new BusinessException(errMsg);
+        }
+
+        InlongClusterTagEntity entity = CommonBeanUtils.copyProperties(request, InlongClusterTagEntity::new);
+        clusterTagMapper.insert(entity);
+        LOGGER.info("success to save cluster tag={} by user={}", request, operator);
```
Comment (kipshi) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:115]: some use Logger.debug , other use Logger.info,  should be unity

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java (hunk 1 above)
Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:102]: OK.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java (hunk 2 above)
Comment (healchow) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:50]: Put them in the under APIs, which will be searched conveniently.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java (hunk 3 above)
Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:115]: Nop, the write operation needs more info logs.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java [hunk 4]
```diff
@@ -47,14 +50,53 @@
...
 
     @Autowired
     private InlongClusterService clusterService;
 
-    @PostMapping(value = "/save")
+    @PostMapping(value = "/cluster/tag/save")
+    @ApiOperation(value = "Save cluster tag")
+    @OperationLog(operation = OperationType.CREATE)
+    @RequiresRoles(value = UserRoleCode.ADMIN)
+    public Response<Integer> saveTag(@RequestBody ClusterTagRequest request) {
```
Comment (gong) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:63]: lost `@Valid`

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java [hunk 5]
```diff
@@ -47,14 +50,53 @@
...
+    @ApiOperation(value = "List cluster tags")
+    public Response<PageInfo<ClusterTagResponse>> listTag(@RequestBody ClusterTagPageRequest request) {
+        request.setCurrentUser(LoginUserUtils.getLoginUserDetail().getUsername());
+        return Response.success(clusterService.listTag(request));
+    }
+
+    @PostMapping(value = "/cluster/tag/update")
+    @OperationLog(operation = OperationType.UPDATE)
+    @ApiOperation(value = "Update cluster tag")
+    public Response<Boolean> updateTag(@RequestBody ClusterTagRequest request) {
```
Comment (gong) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:85]: lost `@Valid`

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java [hunk 6]
```diff
@@ -63,28 +105,28 @@ public Response<Integer> save(@RequestBody InlongClusterRequest request) {
...
     @ApiOperation(value = "List clusters")
     public Response<PageInfo<InlongClusterInfo>> list(@RequestBody InlongClusterPageRequest request) {
         return Response.success(clusterService.list(request));
     }
 
-    @PostMapping(value = "/update")
+    @PostMapping(value = "/cluster/update")
     @OperationLog(operation = OperationType.UPDATE)
     @ApiOperation(value = "Update cluster")
     public Response<Boolean> update(@RequestBody InlongClusterRequest request) {
```
Comment (gong) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:124]: lost `@Valid`

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java (hunk 5 above)
Comment (healchow) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongClusterController.java:85]: Good, done.

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongClusterTagEntityMapper.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.apache.ibatis.annotations.Param;
+import org.apache.inlong.manager.common.pojo.cluster.ClusterTagPageRequest;
+import org.apache.inlong.manager.dao.entity.InlongClusterTagEntity;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface InlongClusterTagEntityMapper {
```
Comment (kipshi) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongClusterTagEntityMapper.java:28]: make some unit test if possible

Comment (healchow) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/InlongClusterTagEntityMapper.java:28]: I will add more logic and unit tests later.
This was used for the inlong-dashboard first.

**Review by kipshi — APPROVED** [review]

**Pull request merged**